public/javascripts/controllers/buyer-ctrl.js

```diff
@@ -227,6 +227,89 @@ angular.module('jasmic.controllers')
                 $scope.demand.de_until = moment().add(7, 'days').toDate();
             };
 
+            $scope.editDemandForm = function (obj) {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                $scope.demand = obj;
+
+                //for some reason the dates needed to be passed through MomentJS as seen below to be assigned to the form inputs
```

Comment:
@nick22891 are you saying that once the user selects the date on the form the obj parameter of this function does not contain the corresponding dates?